source/Octopus.Tentacle.Tests.Integration/Support/TentacleBuilder.cs

```diff
@@ -13,13 +14,39 @@
 
 namespace Octopus.Tentacle.Tests.Integration.Support
 {
-    public abstract class TentacleBuilder<T> where T : TentacleBuilder<T>
+    public interface ITentacleBuilder
+    {
+        ITentacleBuilder WithRunTentacleEnvironmentVariable<TValue>(string environmentVariable, TValue value);
+        ITentacleBuilder WithHomeDirectory(TemporaryDirectory homeDirectory);
+    }
+
+    public abstract class TentacleBuilder<T> : ITentacleBuilder
+        where T : TentacleBuilder<T>
     {
         protected string? ServerThumbprint;
         protected string? TentacleExePath;
         protected string CertificatePfxPath = Certificates.TentaclePfxPath;
         protected string TentacleThumbprint = Certificates.TentaclePublicThumbprint;
-        private readonly Regex listeningPortRegex = new Regex(@"^listen:\/\/.+:(\d+)\/");
+        
+        readonly Regex listeningPortRegex = new (@"^listen:\/\/.+:(\d+)\/");
+        readonly Dictionary<string, string> runTentacleEnvironmentVariables = new();
+
+        TemporaryDirectory? homeDirectory;
+
+        protected TemporaryDirectory HomeDirectory
```

Comment:
If we had this all in 1 class, then we would treat this like all the other builders (i.e. null/override and then ensure it exists during `Build`).

So this would get cleaned up if we make this 1 class.

Reply:
I'm just resolving these as it is adding extra noise and things to read not directly related to this PR